packages/amplication-client/src/Workspaces/hooks/useProjectSelector.ts

```diff
@@ -129,17 +131,25 @@ const useProjectSelector = (
     workspace,
   ]);
 
+  const savedProjectId = localStorage.getItem(CURRENT_PROJECT_ID);
+
+  const savedCurrentProject = useMemo(
+    () => projectsList.find((project_) => project_.id === savedProjectId),
+    [projectsList, savedProjectId]
+  );
+
   useEffect(() => {
     if (
-      !project ||
       !projectsList.length ||
       projectListData.projects.length !== projectsList.length
     )
       return;
 
-    const selectedProject = projectsList.find(
-      (projectDB: models.Project) => projectDB.id === project
-    );
+    const selectedProject =
+      projectsList.find(
+        (projectDB: models.Project) => projectDB.id === project
+      ) ||
+      (!!workspaceUtil && savedCurrentProject);
```

Comment:
instead of using a new memo for savedCurrentProject, you can use here the savedProjectId to find the project.

      projectsList.find(
        (projectDB: models.Project) => projectDB.id === project || savedProjectId);

Also, we need to wrap the selectedProject with useMemo (I understand it was like that before), so - something like:
```
const selectedProject = useMemo(()=>{
  return projectsList.find(
        (projectDB: models.Project) => projectDB.id === project || savedProjectId);
},[...)
```

Reply:
Hi @gitstart ,
I am closing this PR due to inactivity (no response for the comments above). If you wish to work on this, feel free to re-open the PR and add the requested changes.
Thanks and sorry.

Reply:
Requested changes have been addressed here https://github.com/amplication/amplication/pull/6544. Thanks. cc @mulygottlieb @yuval-hazaz

Reply:
Thanks @gitstart! The team will review the new PR in the coming days.